homework-2/index.ts

```diff
@@ -0,0 +1,101 @@
+import express, { NextFunction, Response, Request } from 'express';
+import { v4 } from 'uuid';
+import Joi from '@hapi/joi';
+
+const app = express();
+app.use(express.json());
+
+/* Type declaration and validation schema for User */
+type User = {
+  id: string;
+  login: string;
+  password: string;
+  age: number;
+  isDeleted: boolean;
+};
+const users: Array<User> = [];
+
+const userSchema = Joi.object({
+  login: Joi.string().required(),
+  password: Joi.string().required().alphanum(),
+  age: Joi.number().required().min(4).max(130),
+});
+type UserSchema = typeof userSchema;
+
+const userValidator = (schema: UserSchema) => {
+  return (req: Request, res: Response, next: NextFunction) => {
+    const { error } = schema.validate(req.body);
+
+    error?.isJoi ? res.status(400).json(error.message) : next();
+  }
+}
+
+/* CRUD operations for Users */
+app.post('/users', userValidator(userSchema), (req, res) => {
+  const user = req.body as User;
+
+  user.id = v4();
+  user.isDeleted = false;
+
+  users.push(user);
+  console.info(users);
+
+  res.send(user);
```

Comment:
Maybe it makes sense to send 201 status code here

Reply:
Also, it'd be nice to have "Location" header with /users/<created_id> value for restful compliance